tests/registry/registry.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package registry
+
+import (
+	"github.com/onsi/ginkgo/v2"
+
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+type TestFunc func(t ginkgo.FullGinkgoTInterface, tn workload.TestNetwork) error
+
+type namedTest struct {
+	Fnc  TestFunc
+	Name string
+}
+type Registry struct {
+	tests []namedTest
+}
+
+func (r *Registry) Add(name string, f TestFunc) {
+	r.tests = append(r.tests, namedTest{Fnc: f, Name: name})
+}
+
+func (r *Registry) List() []namedTest {
+	if r == nil {
+		return []namedTest{}
+	}
+	return r.tests
+}
+
+// we need to pre-register all the test registries that are created externally in order to comply with the ginko execution order.
+// i.e. the global `var _ = ginkgo.Describe` used in the integration/e2e tests need to have this field populated before the iteration
+// over the top level nodes.
```

Comment:
Can we avoid this by setting this in a global variable / setting via `init` function so that it's identical during test construction and execution?

Reply:
I think that it is identical during test construction and execution.

Not sure how to implement that so that it would work `ginkgo` otherwise, as `ginkgo` is managing these in globals as well ( but keep them internal )